src/program.ts

```diff
@@ -3646,17 +3686,26 @@ export class InterfacePrototype extends ClassPrototype { // FIXME
       true
     );
   }
+  /** Convert a class to an interface when an interface extends a class */
+  static fromClassPrototype(cP: ClassPrototype): InterfacePrototype {
+    const cDecl = <ClassDeclaration> cP.declaration;
+    const iDecl = new InterfaceDeclaration();
+    iDecl.typeParameters = cDecl.typeParameters;
+    iDecl.extendsType = cDecl.extendsType;
+    iDecl.implementsTypes = cDecl.implementsTypes;
+    iDecl.members = cDecl.members;
+    return new InterfacePrototype(cP.name, cP.parent, iDecl, cP.decoratorFlags);
+  }
 }
 
 /** A resolved interface. */
 export class Interface extends Class { // FIXME
-
   /** Constructs a new interface. */
   constructor(
     nameInclTypeParameters: string,
     prototype: InterfacePrototype,
-    typeArguments: Type[] = [],
-    base: Interface | null = null
+    typeArguments: Type[] | null = [],
+    base: Interface | Class | null = null // interface can extend classes in typescript
```

Comment:
There are no union types when compiling this as AS

Reply:
Yeah it should just be class.